### PR TITLE
Cleanup code in Node.h/.cpp

### PR DESCRIPTION
**Pull request by dumganhar**
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/core/scene-graph/Node.h [hunk 1]
```diff
@@ -357,10 +357,10 @@ class Node : public BaseNode {
     /**
      * @en Inversely transform a point from world coordinate system to local coordinate system.
      * @zh 逆向变换一个空间点，一般用于将世界坐标转换到本地坐标系中。
-     * @param out The result point in local coordinate system will be stored in this vector
      * @param p A position in world coordinate system
+     * @return The result point in local coordinate system will be stored in this vector
      */
-    void inverseTransformPoint(Vec3 &out, const Vec3 &p);
+    Vec3 inverseTransformPoint(const Vec3 &p);
```
Comment (minggo) [inline native/cocos/core/scene-graph/Node.h:363]: Why change it. It is different from TS implementation.

Comment (dumganhar) [reply native/cocos/core/scene-graph/Node.h:363]: There are two main reasons:

1. `Vec3& out` is not user friendly for c++ developers, c++ has stack for returning a temporary variable, but JS objects are allocated on heap, so JS needs a `out` parameter for returning value without generating GC objects.

2. JS binding could not bind a function with a `out` parameter, the original `inverseTransformPoint` auto binding code will not work properly.

Actually, we need to bind `inverseTransformPoint` manually. I will do it later.

Comment (minggo) [reply native/cocos/core/scene-graph/Node.h:363]: Ok. Will modify in this PR, right?

Comment (dumganhar) [reply native/cocos/core/scene-graph/Node.h:363]: No, I will create a new issue for that.

Comment (dumganhar) [reply native/cocos/core/scene-graph/Node.h:363]: https://github.com/cocos/cocos-engine/issues/11570

**Review by minggo — APPROVED** [review]

**Pull request merged**